src/librustc_mir/build/cfg.rs

```diff
@@ -48,9 +49,13 @@ impl<'tcx> CFG<'tcx> {
                            block: BasicBlock,
                            source_info: SourceInfo,
                            extent: CodeExtent) {
-        self.push(block, Statement {
```

Comment:
Can't you pass the `tcx` here instead of using TLS? Using TLS like that is kind of an anti-pattern.

Reply:
I thought I considered this and rejected it (based on a fear of having to thread `tcx` up a long method chain), but looking at it now I don't know why I came to that conclusion, since the `Builder` in `scope.rs` carries a `hair::cx::Cx`...

Will fix.